MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1098,88 +1131,95 @@ void EnggDiffractionPresenter::doFocusRun(const std::string &dir,
                                           const std::string &specNos,
                                           const std::string &dgFile) {
 
-  g_log.notice() << "Generating new focusing workspace(s) and file(s) into "
-                    "this directory: "
-                 << dir << std::endl;
-
-  // TODO: this is almost 100% common with doNewCalibrate() - refactor
-  EnggDiffCalibSettings cs = m_view->currentCalibSettings();
-  Mantid::Kernel::ConfigServiceImpl &conf =
-      Mantid::Kernel::ConfigService::Instance();
-  const std::vector<std::string> tmpDirs = conf.getDataSearchDirs();
-  // in principle, the run files will be found from 'DirRaw', and the
-  // pre-calculated Vanadium corrections from 'DirCalib'
-  if (!cs.m_inputDirCalib.empty() && Poco::File(cs.m_inputDirCalib).exists()) {
-    conf.appendDataSearchDir(cs.m_inputDirCalib);
-  }
-  if (!cs.m_inputDirRaw.empty() && Poco::File(cs.m_inputDirRaw).exists()) {
-    conf.appendDataSearchDir(cs.m_inputDirRaw);
-  }
+  while (!g_abortThread) {
```

Comment:
The stop button is working also well for me on linux. But the focus process gets now into an infinite loop of focusing the same run forever. I think that in this `while` what you mean is an `if? The loop is done in the worker. Here this is about one run to focus. An alternative to putting all the method inside a `while` or `if` or similar is to put at the beginning:

```
if (g_abortThread)
    return
```